src/shared/components/Posts/components/SinglePost/SinglePost.tsx

```diff
@@ -0,0 +1,53 @@
+import { FC } from 'react';
+import { IPostResponse } from '../../interfaces/IPostsResponse';
+//components
+import { PostsComment } from '../PostsComment';
+//styles
+import styles from './SinglePost.module.scss';
+//mui
+import { CardContent, Typography } from '@mui/material';
+
+interface ISinglePost {
+    post: IPostResponse;
+    showComments?: number;
+}
+
+const SinglePost: FC<ISinglePost> = (props) => {
+    const { post, showComments } = props;
+
```

Comment:
try if (post) return null I think you aren't going to need check for post.comments and post.user